common/resource/resourceImpl.go

```diff
@@ -297,7 +303,9 @@ func New(
 			logger,
 			params.InstanceID,
 		),
-		rpcFactory: params.RPCFactory,
+		rpcFactory:      params.RPCFactory,
+		isolationGroups: params.isolationGroupState,
```

Comment:
I think we need some methods to access these fields

Reply:
I don't love getters like this, but good point